### DISCUSSION TITLE
artifact url not processing js files in plugin:

### DISCUSSION THREAD

**Discussion by matthewstick** [craftcms/cloud#17] [Feature Requests & Enhancements]
We have a form here, generated by the popular formie plugin:
https://treasureisland.com/entertainment/bars-lounges/the-cove-bar-arcade/the-cove-bar-arcade-disclaimer

Note that the datepicker does not work.  Nor does the signature panel work (you can draw on the field with your mouse like a signature).  These work fine in a non-craft cloud environment.

I suspect the issue is that these two related js libraries are not being loaded.  From the logs, I get file not found:

https://cdn.craft.cloud/e5c2b9ea-b8f1-4463-86d9-face683786c7/builds/9bf10285-4fbe-47bd-a7f4-92bea13c5ddb/cpresources/verbb-formie-web-assets-frontend-dist-js/fields/signature.js?v=1712724063

https://cdn.craft.cloud/e5c2b9ea-b8f1-4463-86d9-face683786c7/builds/9bf10285-4fbe-47bd-a7f4-92bea13c5ddb/cpresources/verbb-formie-web-assets-frontend-dist-js/fields/date-picker.js?v=1712724063

Note signature.js and date-picker.js are not in our front end code anywhere.  They are part of the formie plugin.  I have no control over these two files.  There are a lot more files actually that formie uses.  These are just two.  These files are located here: vendor/verbb/formie/src/web/assets/frontend/

It seems that Craft Cloud is almost making the link correct, but it does not.

I suspect the problem is one of these:
A: Craft Cloud does try to take into account JS/CSS/etc "artifact"files that are loaded from plugins. But there is a bug somewhere.  And thus the links are bad.
B: Craft Cloud does NOT take into account JS/CSS/etc "artifact" files that are loaded from plugins.  And Craft Cloud does not support plugins.  And that plugin developer need to remake their plugins to get craft cloud approved.

If the answer is A or something else, can you please fix/address?

If the answer is B, this is a pretty big deal breaker for us.  We spent a lot of time porting the app, only to find out that plugins don't work.  It would be helpful to know in advance if a plugin is craft cloud approved or not.  I'm emailing this discussion ticket to formie as well as an fyi.

(As a side note, I think the artifactUrl method is tough to use and is causing some issues for us, such as this.  I understand you really made something nice here - but wish artifact url could just go away and it would "just magically work".)

Please let us know what you think.  We might have to pull all of our sites we migrated and were planning to migrate that use formie off of craft cloud if this is a bigger issue.

Many thanks!

**Comment by timkelty** [comment]
This is just a problem with how the plugin is registering those assets.

We'll work with the developer to get it patched asap.

**Comment by matthewstick** [comment]
Glorious !

**Comment by timkelty** [comment]
Should be fixed by: https://github.com/verbb/formie/pull/1859

**Comment by matthewstick** [comment]
Love it @timkelty .  It works.  Thanks so much.
